`cargo test` fails and leaves some files changed

Running `cargo test` in a checkout of this repo fails when I run it locally:
```
$ cargo test
   Compiling ui_test v0.4.0 (/home/r/src/rust/lib-crates/ui_test)
    Finished test [unoptimized + debuginfo] target(s) in 3.65s
     Running unittests src/lib.rs (target/debug/deps/ui_test-a70863804512cefe)

running 13 tests
test parser::tests::parse_missing_level ... ok
test parser::tests::parse_slash_slash_at ... ok
test parser::tests::missing_colon_fail ... ok
test parser::tests::parse_simple_comment ... ok
test parser::tests::parse_slash_slash_at_fail ... ok
test parser::tests::parse_x86_64 ... ok
test tests::issue_2156 ... ok
test tests::missing_implicit_warn_pattern ... ok
test tests::duplicate_pattern ... ok
test tests::missing_pattern ... ok
test tests::find_pattern ... ok
test tests::missing_warn_pattern ... ok
test parser::tests::parse_regex_error_pattern ... ok

test result: ok. 13 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/integration.rs (target/debug/deps/integration-dfceebd5da778812)

Running `integrations` tests in mode pass
...
test result: ok. 3 tests passed, 0 ignored, 0 filtered out


Running `integrations` tests in mode fail
F
tests/integrations/basic-fail/Cargo.toml FAILED:
command: "cargo" "test" "--color" "never" "--jobs" "1" "--target-dir" "target" "--manifest-path" "tests/integrations/basic-fail/Cargo.toml" "--" "--test-threads" "1"

actual output differed from expected
--- tests/integrations/basic-fail/Cargo.stderr
+++ <stderr output>
... 97 lines skipped ...
 
 test result: FAIL. 3 tests failed, 0 tests passed, 0 ignored, 0 filtered out
-error: test failed, to rerun pass '--test ui_tests'
+error: test failed, to rerun pass `--test ui_tests`
+
+Caused by:
+  process didn't exit successfully: `$DIR/target/debug/ui_tests-107eeeee069799c8 --test-threads 1` (exit status: 1)
 


full stderr:
   Compiling basic_fail v0.1.0 (/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail)
    Finished test [unoptimized + debuginfo] target(s) in 2.31s
     Running unittests src/lib.rs (target/debug/deps/basic_fail-20ffe916eeb1df79)
     Running tests/ui_tests.rs (target/debug/deps/ui_tests-107eeeee069799c8)
   Compiler flags: ["--error-format=json", "--edition=2021"]
   Building test dependencies...
tests/actual_tests/bad_pattern.rs ... FAILED
tests/actual_tests/filters.rs ... FAILED
tests/actual_tests/foomp.rs ... FAILED

tests/actual_tests/bad_pattern.rs FAILED:
command: "rustc" "--error-format=json" "--edition=2021" "--extern" "basic_fail=/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug/deps/libbasic_fail-e8149fec574d4432.rmeta" "-L" "/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug" "-L" "/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug/deps" "tests/actual_tests/bad_pattern.rs"

substring `miesmätsched types` not found in stderr output
expected because of pattern here: tests/actual_tests/bad_pattern.rs:5

There were 1 unmatched diagnostics at tests/actual_tests/bad_pattern.rs:4
    Error: mismatched types

full stderr:
error[E0308]: mismatched types
 --> tests/actual_tests/bad_pattern.rs:4:9
  |
4 |     add("42", 3);
  |     --- ^^^^ expected `usize`, found `&str`
  |     |
  |     arguments to this function are incorrect
  |
note: function defined here
 --> /home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/src/lib.rs:1:8
  |
1 | pub fn add(left: usize, right: usize) -> usize {
  |        ^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.



tests/actual_tests/filters.rs FAILED:
command: "parse comments"

Could not parse comment in tests/actual_tests/filters.rs:1 because invalid condition `"x86_64"`, expected `on-host`, /[0-9]+bit/ or /target-.*/

full stderr:



tests/actual_tests/foomp.rs FAILED:
command: "rustc" "--error-format=json" "--edition=2021" "--extern" "basic_fail=/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug/deps/libbasic_fail-e8149fec574d4432.rmeta" "-L" "/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug" "-L" "/home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/../../../target/debug/deps" "tests/actual_tests/foomp.rs"

actual output differed from expected
--- tests/actual_tests/foomp.stderr
+++ <stderr output>
... 3 lines skipped ...
 4 |     add("42", 3);
   |     --- ^^^^ expected `usize`, found `&str`
+  |     |
+  |     arguments to this function are incorrect
   |
 note: function defined here
~ --> $DIR/tests/integrations/basic-fail/src/lib.rs:1:8
   |
 1 | pub fn add(left: usize, right: usize) -> usize {
~  |        ^^^ some expected text that isn't in the actual message
 
-error: aborting doo to previous error
+error: aborting due to previous error
 
 For more information about this error, try `rustc --explain E0308`.
 


full stderr:
error[E0308]: mismatched types
 --> tests/actual_tests/foomp.rs:4:9
  |
4 |     add("42", 3);
  |     --- ^^^^ expected `usize`, found `&str`
  |     |
  |     arguments to this function are incorrect
  |
note: function defined here
 --> /home/r/src/rust/lib-crates/ui_test/tests/integrations/basic-fail/src/lib.rs:1:8
  |
1 | pub fn add(left: usize, right: usize) -> usize {
  |        ^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.


FAILURES:
    tests/actual_tests/bad_pattern.rs
    tests/actual_tests/filters.rs
    tests/actual_tests/foomp.rs

test result: FAIL. 3 tests failed, 0 tests passed, 0 ignored, 0 filtered out
error: test failed, to rerun pass `--test ui_tests`

Caused by:
  process didn't exit successfully: `/home/r/src/rust/lib-crates/ui_test/target/debug/deps/ui_tests-107eeeee069799c8 --test-threads 1` (exit status: 1)


FAILURES:
    tests/integrations/basic-fail/Cargo.toml

test result: FAIL. 1 tests failed, 0 tests passed, 0 ignored, 0 filtered out
error: test failed, to rerun pass `--test integration`

Caused by:
  process didn't exit successfully: `/home/r/src/rust/lib-crates/ui_test/target/debug/deps/integration-dfceebd5da778812` (exit status: 1)
```

Furthermore, when it is done, some files have changed:
```diff
diff --git a/tests/integrations/basic-bin/Cargo.lock b/tests/integrations/basic-bin/Cargo.lock
index b58f85f..138b740 100644
--- a/tests/integrations/basic-bin/Cargo.lock
+++ b/tests/integrations/basic-bin/Cargo.lock
@@ -500,7 +500,7 @@ dependencies = [
 
 [[package]]
 name = "ui_test"
-version = "0.3.1"
+version = "0.4.0"
 dependencies = [
  "bstr",
  "cargo_metadata",
diff --git a/tests/integrations/basic-fail-mode/Cargo.lock b/tests/integrations/basic-fail-mode/Cargo.lock
index f64d5cb..dc0f81e 100644
--- a/tests/integrations/basic-fail-mode/Cargo.lock
+++ b/tests/integrations/basic-fail-mode/Cargo.lock
@@ -501,7 +501,7 @@ dependencies = [
 
 [[package]]
 name = "ui_test"
-version = "0.3.1"
+version = "0.4.0"
 dependencies = [
  "bstr",
  "cargo_metadata",
diff --git a/tests/integrations/basic-fail/Cargo.lock b/tests/integrations/basic-fail/Cargo.lock
index 3b4337f..1848885 100644
--- a/tests/integrations/basic-fail/Cargo.lock
+++ b/tests/integrations/basic-fail/Cargo.lock
@@ -500,7 +500,7 @@ dependencies = [
 
 [[package]]
 name = "ui_test"
-version = "0.3.1"
+version = "0.4.0"
 dependencies = [
  "bstr",
  "cargo_metadata",
diff --git a/tests/integrations/basic/Cargo.lock b/tests/integrations/basic/Cargo.lock
index 20d87bb..9a47117 100644
--- a/tests/integrations/basic/Cargo.lock
+++ b/tests/integrations/basic/Cargo.lock
@@ -501,7 +501,7 @@ dependencies = [
 
 [[package]]
 name = "ui_test"
-version = "0.3.1"
+version = "0.4.0"
 dependencies = [
  "bstr",
  "cargo_metadata",
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -17,7 +17,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.63.0
+        toolchain: stable
         profile: minimal
         components: rustfmt
         override: true
@@ -31,7 +31,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.63.0
+        toolchain: stable
         profile: minimal
         components: clippy
         override: true
@@ -57,7 +57,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.63.0
+        toolchain: stable
         profile: minimal
         override: true
     - name: Build

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -15,12 +15,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
+    - uses: dtolnay/rust-toolchain@stable
       with:
-        toolchain: stable
-        profile: minimal
         components: rustfmt
-        override: true
     - name: Format
       run: cargo fmt --check
 
@@ -29,12 +26,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
+    - uses: dtolnay/rust-toolchain@stable
       with:
-        toolchain: stable
-        profile: minimal
         components: clippy
-        override: true
     - name: Clippy
       run: cargo clippy -- -D warnings
   
@@ -43,11 +37,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: nightly
-          profile: minimal
-          override: true
+      - uses: dtolnay/rust-toolchain@nightly
       - name: Check semver
         uses: obi1kenobi/cargo-semver-checks-action@v1
 
@@ -55,11 +45,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        profile: minimal
-        override: true
+    - uses: dtolnay/rust-toolchain@stable
     - name: Build
       run: cargo build --verbose
     - name: Run tests

--- a/tests/integration.rs
+++ b/tests/integration.rs
@@ -48,6 +48,10 @@ fn run(name: &str, mode: Mode) -> Result<()> {
     config.stderr_filter("( +Running [^(]+).*", "$1");
     config.stderr_filter(" *Blocking waiting for.*\n", "");
     config.stderr_filter(" *(Compiling|Downloaded|Downloading) .*\n", "");
+    config.stderr_filter(
+        "( *process didn't exit successfully: `[^-]+)-[0-9a-f]+",
+        "$1-HASH",
+    );
     // The order of the `/deps` directory flag is flaky
     config.stderr_filter("/deps", "");
     config.stderr_filter(

--- a/tests/integrations/basic-fail/Cargo.stderr
+++ b/tests/integrations/basic-fail/Cargo.stderr
@@ -97,4 +97,7 @@ FAILURES:
     tests/actual_tests/foomp.rs
 
 test result: FAIL. 3 tests failed, 0 tests passed, 0 ignored, 0 filtered out
-error: test failed, to rerun pass '--test ui_tests'
+error: test failed, to rerun pass `--test ui_tests`
+
+Caused by:
+  process didn't exit successfully: `$DIR/target/debug/ui_tests-9c541b861fc6e519 --test-threads 1` (exit status: 1)

--- a/tests/integrations/basic-fail/Cargo.stderr
+++ b/tests/integrations/basic-fail/Cargo.stderr
@@ -100,4 +100,4 @@ test result: FAIL. 3 tests failed, 0 tests passed, 0 ignored, 0 filtered out
 error: test failed, to rerun pass `--test ui_tests`
 
 Caused by:
-  process didn't exit successfully: `$DIR/target/debug/ui_tests-9c541b861fc6e519 --test-threads 1` (exit status: 1)
+  process didn't exit successfully: `$DIR/target/debug/ui_tests-HASH --test-threads 1` (exit status: 1)

